dotCMS/src/main/java/com/dotmarketing/util/ExportStarterUtil.java

```diff
@@ -54,26 +64,60 @@ public ExportStarterUtil() {
         
         outputDirectory = new File(ConfigUtils.getBackupPath() + File.separator + "backup_" + System.currentTimeMillis());
         outputDirectory.mkdirs();
-       
-        
     }
     
-    public File createStarterWithAssets() throws FileNotFoundException, IOException {
-        
+    public File createStarterWithAssets() throws IOException {
         moveAssetsToBackupDir();
         return createStarterData() ;
-        
     }
     
     public File createStarterData() {
         createXMLFiles() ;
-        
         return outputDirectory;
-        
-        
     }
-    
-    
+
+    /**
+     * Manipulates contents of a directory and put them through a given output stream not without first compressing them
+     * to a generated zip file.
+     *
+     * @param output given output stream
+     * @param withAssets flag telling to include assets or not
+     * @param download flag telling to download as a file or as a stream
+     * @return file zip file
+     * @throws IOException
+     */
+    public Optional<File> zipStarter(final OutputStream output, final boolean withAssets, final boolean download)
```

Comment:
Tests

Reply:
This is tested from the actual postman test.